pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java

```diff
@@ -55,7 +65,18 @@ public TimeGranularitySpec(@Nonnull DataType dataType, @Nonnull TimeUnit timeTyp
     _timeType = timeType;
     _name = name;
   }
+  
+  public TimeGranularitySpec(@Nonnull DataType dataType, @Nonnull TimeUnit timeType, @Nonnull String timeFormat,
+      @Nonnull String name) {
+    Preconditions.checkNotNull(timeType);
+    Preconditions.checkNotNull(name);
 
+    _dataType = dataType.getStoredType();
+    _timeType = timeType;
+    _name = name;
+    _timeFormat = timeFormat;
```

Comment:
Should we not do some validation on the time format here? (should start with SImpleDateFormat, and should have a valid string that is parsable by DateFormat?